faf-commons-data/src/main/java/com/faforever/commons/replay/shared/LuaData.java

```diff
@@ -1,11 +1,86 @@
 package com.faforever.commons.replay.shared;
 
+import java.util.HashMap;
 import java.util.Map;
 
 public sealed interface LuaData {
+  record Nil() implements LuaData {}
   record Number(float value) implements LuaData {}
   record String(java.lang.String value) implements LuaData {}
-  record Nil() implements LuaData {}
-  record Table(Map<java.lang.String, LuaData> value) implements LuaData {}
   record Bool(boolean value) implements LuaData {}
+
+  record Table(Map<java.lang.String, LuaData> value) implements LuaData {
+
+    public void removeKey(java.lang.String key) {
+      value.remove(key);
+    }
+
+    public Boolean getBool(java.lang.String key) {
+      if (value.get(key) instanceof LuaData.Bool (boolean bool)) {
+        return bool;
+      }
+
+      return null;
+    }
+
+    public java.lang.String getString(java.lang.String key) {
+      if (value.get(key) instanceof LuaData.String (java.lang.String string)) {
+        return string;
+      }
+
+      return null;
+    }
+
+    public Float getFloat(java.lang.String key) {
+      if (value.get(key) instanceof LuaData.Number (float number)) {
+        return number;
+      }
+
+      return null;
+    }
+
+    public Integer getInteger(java.lang.String key) {
+      if (value.get(key) instanceof LuaData.Number (float number)) {
+        return  (Integer) (int) Math.floor(number);
+      }
+
+      return null;
+    }
+
+    public LuaData.Table getTable(java.lang.String key) {
+      if (value.get(key) instanceof LuaData.Table table) {
+        return  table;
+      }
+
+      return null;
+    }
+
+    public Map<java.lang.String, java.lang.String> toMap() {
```

Comment:
What is this method used for?

Reply:
It is used for all the 'other' game options that originate from mods. The idea is that we do not or as little as possible expose the LuaData record so that other programs that make use of the library can use the data types that they know. In this example that would be a map.

Reply:
But the LuaData objects are part of the public API so why not expose it?

Additionally in the case of a table all information is lost so it ends up not being that helpful. And the nil value if that was the case should probably just be translated to the java null so it is truly more native. But I think avoiding exposing luaData is a bit unnecessary since you then lose all the typing information we just put effort into parsing.